client/my-sites/plans/jetpack-plans/product-store/hooks/use-item-lightbox.ts

```diff
@@ -1,24 +1,38 @@
 import { useCallback, useMemo, useState } from 'react';
 import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { EXTERNAL_PRODUCTS_LIST } from '../../constants';
+import slugToSelectorProduct from '../../slug-to-selector-product';
 import { SelectorProduct } from '../../types';
+import { sanitizeLocationHash } from '../utils/sanitize-location-hash';
 
 export const useItemLightbox = () => {
-	const [ currentItem, setCurrentItem ] = useState< SelectorProduct | null >( null );
+	const item = slugToSelectorProduct( sanitizeLocationHash( window.location.hash ) );
+
+	const [ currentItem, setCurrentItem ] = useState< SelectorProduct | null >( item );
```

Comment:
Can we move this logic to a callback to `useState` to avoid it getting computed everytime?

```suggestion
	const [ currentItem, setCurrentItem ] = useState( () => {
		return slugToSelectorProduct( sanitizeLocationHash( window.location.hash ) );
	} );
```

Reply:
Yes I agree.. this should be inside useState 👍 .